### ISSUE TITLE
what is the visibility of setPropertyValue?

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/axon#297]
Property.js, added by @samreid:

```js
   * @protected - for overriding only
   */
  setPropertyValue( value ) {
```

TinyProperty.js, added by @jonathanolson:

```js
   * @public
   *
   * @param {*} value
   */
  setPropertyValue( value ) {
```

Note the different visibility annotations.  Is there a reason why they are different? If not, which is correct?    I have an immediate need to use Property.setPropertyValue in Natural Selection.

**Comment by samreid** [comment]
They are both correct because they have different use cases.  If it is important for simulation code to call `Property.setPropertyValue`, we can convert it to `@public`.  What is the use case for Natural Selection?  Is there a better way to proceed?

**Comment by pixelzoom** [comment]
> They are both correct because they have different use cases. 

Can you be more specific? What is the use case that requires TinyProperty.setInitialValue to be public?

> What is the use case for Natural Selection? Is there a better way to proceed?

Here's the use case in Natural Selection: 

The initial value of Gene `dominanteAlleleProperty` is set by the (public!) `introMutations` and `labMutations` query parameters, for _Intro_ and _Lab_ screen respectively.   But it's not possible to set the initial value of `dominanteAlleleProperty` via its constructor, because the query parameters can only be parsed after the GenePool (and Genes) have been created. The set of Genes in the GenePool determines which allele characters may appear in the query parameter values.  

So here's the startup sequence for the Intro screen (Lab screen is similar):

1. Instantiate `GenePool` with a set of `Gene` instances, each of which has a `dominanteAlleleProperty` with default value `null` (meaning no dominant allele).

2. Parse `introMutations` by iterating over Genes in the GenePool to identify valid allele abbreviations. For example, if the GenePool contains the Fur gene, then 'F' and 'f' are valid characters.  If there are any errors in the query parameter, notify the user via QueryStringMachine.addWarning, which results in a Dialog.

3. Based on which allele abbreviation are in the `introMutations` value, set the initial value of a Gene's `dominanteAlleleProperty` using `setInitialValue`.  For example, for `introMutations=Fe`, the mutation allele (brown fur) is dominant for the Fur gene, the normal allele (straight ears) is dominant for the Ears gene, and the dominant allele for Teeth gene remains `null` because it is not represented in the query parameter value.

**Comment by samreid** [comment]
>. Can you be more specific? What is the use case that requires TinyProperty.setInitialValue to be public?

I presume you meant `TinyProperty.setPropertyValue`.  There is other work that needs to be done between setting the value and notifying listeners.

// RichText.js
```js
this.textProperty.setPropertyValue( text );
this.rebuildRichText();

this.textProperty.notifyListeners( oldText );
```

// Text.js
```js
this.textProperty.setPropertyValue( text );
this._cachedRenderedText = null;

const stateLen = this._drawables.length;
for ( let i = 0; i < stateLen; i++ ) {
  this._drawables[ i ].markDirtyText();
}

this.invalidateText();

this.textProperty.notifyListeners( oldText );
```

// Node.js
```js
// no paint or invalidation changes for now, since this is only handled for the mouse
this.pickableProperty.setPropertyValue( pickable );

this._picker.onPickableChange( oldPickable, pickable );
if ( assertSlow ) { this._picker.audit(); }
// TODO: invalidate the cursor somehow? #150

this.pickableProperty.notifyListeners( oldPickable );
```

Brainstorming options to discuss:
A. Make `Property.setPropertyValue` public
B. Change the startup sequence so that `dominanteAlleleProperty` is only constructed when we know what the initial value should be
C. Just use `Property.setValue` and deal with notifications going out

**Comment by pixelzoom** [comment]
Sorry, NS is not using `setPropertyValue`, it is indeed using `setInitialValue`.  I had this confused with an issue issue that I had yet to create for this TODO in NS parseInitialPopulation.js:

```js
      //TODO setInitialValue is annotated @public (PhetioStateEngine), can we change to @public?
      gene.dominantAlleleProperty.setInitialValue( null );
```

The issue is https://github.com/phetsims/axon/issues/299

Re this issue, I guess we can go ahead and close.  I created this issue while discussing the difference between Property and TinyProperty with another developer, and we couldn't understand why they were different.  That may happen again, but I guess we'll deal with it then.  Closing.
